### PR TITLE
Bug fix - unsupported CellValues.String DataType 

### PR DESCRIPTION
**Pull request by Gallardo13**
Fixed bug with missing cell values of excel file, which had CellValues.String DataType.

### CODE REVIEW

**Comment by Pankraty** [comment]
Alexandr, thanks for participation!

A couple of remarks:
1. You could create a pull request from `Gallardo13:develop` directly; no need to merge it to your `develop` before.
2. It is essential for bug fixes to have unit tests which would fail without your fixes and succeed afterwards. Without it, it is impossible to guess what was the bug on the first place, and there is a risk to have a regression after any any kind of refactoring. So please add a unit test.

**Comment by Gallardo13** [comment]
Hello, Alexei!

I committed a little unit test, which will failed without my changes at the latest fix. 
Unfortunately, commit also have three files, which weren't changed, but git thinks what there are changes at there. 
I believe, that this happens, because I use Visual Studio for Mac. 
And when I am trying to revert commit, git reverts just two files, which really have diffs (file with tests and new excel template for test running). How can we solve this problem?

**Comment by Pankraty** [comment]
How did you create this file? If it is saved with Excel the cell type is converted to `InlineString` which is fine. According to specification, CellType=String is only used for cell containing formula text:
image

See also https://stackoverflow.com/questions/4740179/whats-the-difference-between-c-t-str-and-cis-in-office-open-xml

**Comment by Gallardo13** [comment]
Thanks for xml files reverting!

With desktop excel I couldn't create file with this problem. 
I faced with this problem when I tried to process programmatically created excel file in our company. 
There was applied to some cells an OpenXmlAttribute to set type (like this ( "t", null, "str")) at automatically generated file from database content. It's a really legacy code and all that I tried is to change OpenXmlAttribute type to ("t", "null", "s") and ("t", "null", "inlineStr") to understand what will change. And it crashed all the process. I don't know reasons, why developers, who wrote this code some years ago, decided to use String Data type, but now a lot of code around depends to it. I made our own version of ClosedXML library to solve this problem and decided to make pull request here, because some developers will be able to face this problem in the future too.

**Comment by Gallardo13** [comment]
Done.

About conditions - I wrote them in the same style, like was in the code around. Maybe some refactoring for this file are needed?

**Comment by Pankraty** [comment]
We migrated to C# 7 not long ago, so most of the code does not use new language features. It will be refactored eventually.

**Comment by igitur** [comment]
@Pankraty What exactly caused the changes in https://github.com/Gallardo13/ClosedXML/commit/a71d521dd962024901bc59e6268e57c328edbf09 ? I've seen them before. First I thought it was line endings, but no.

**Comment by igitur** [comment]
@Gallardo13 After all feedback is completed, please squash all your commits into one commit with a single descriptive commit message.

**Comment by Pankraty** [comment]
> What exactly caused the changes in Gallardo13/ClosedXML@a71d521 ? I've seen them before. First I thought it was line endings, but no.

I've no idea actually. Just reverted the changes.

File: ClosedXML_Tests/Excel/Loading/LoadingTests.cs [hunk 1]
```diff
@@ -378,5 +379,15 @@ public void LoadFormulaCachedValue(string formula, object expectedValueCached, o
...
             }
         }
+
+        [Test]
+        public void CanCorrectLoadWorkbookCellWithStringDataType() 
+        {
+            using (var stream = TestHelper.GetStreamFromResource(TestHelper.GetResourcePath(@"Examples\Loading\CellWithStringDataType.xlsx")))
+            using (var wb = new XLWorkbook(stream))
+            {
+                Assert.AreEqual("String with String Data type", wb.Worksheet(1).Cell("B2").Value);
```
[the remark this answers is not in the document]
Comment (Gallardo13) [reply ClosedXML_Tests/Excel/Loading/LoadingTests.cs]: Francois, Hello!

XLDataType at ClosedXML have just Text type, no differences between String, Inline string and Shared string. To make an assert for cell data type I also should open the document like OpenXML Spreadsheet - is it correct solution?

**Comment by Gallardo13** [comment]
Add feedback was completed

**Comment by igitur** [comment]
Thanks for the contribution. Will be merged for v0.94. Please keep this PR open. We will close it when it's merged.

**Comment by igitur** [comment]
Please rebase this on the latest develop branch. There is a conflict to resolve.

**Comment by Gallardo13** [comment]
Done

**Comment by Gallardo13** [comment]
git rebase upstream/develop made this result 

try doing it again, and the same situation

**Comment by igitur** [comment]
After rebasing, did you push to your repo? You will have to force the push.

**Comment by Gallardo13** [comment]
Trying to rebase and make force push, result still not correct

**Comment by Gallardo13** [comment]
cmd

I did following sequence of actions, maybe it helps to you:
1) command - git rebase upstream/develop
2) resolving conflict at ClosedXML_Tests/Excel/Loading/LoadingTests.cs
3) command - git rebase --continue 
4) again conflict, at resolving I delete all changes at ClosedXML_Tests/Excel/Loading/LoadingTests.cs - they aren't needed
5) command - git rebase --continue
result: No changes - did you forget to use 'git add'?
If there is nothing left to stage, chances are that something else
already introduced the same changes; you might want to skip this patch.
6) command - git rebase --skip
7) git pull
8) git push --force

**Comment by igitur** [comment]
Fixed it in #873

**Pull request closed without merging**